river/command/map.zig

```diff
@@ -148,6 +184,21 @@ fn mappingExists(
     return null;
 }
 
+/// Returns the index of the SwitchMapping with matching switch_type and switch_state, if any.
+fn switchMappingExists(
+    switch_mappings: *std.ArrayListUnmanaged(SwitchMapping),
+    switch_type: Switch.Type,
+    switch_state: Switch.State,
+) ?usize {
+    for (switch_mappings.items) |mapping, i| {
+        if (std.meta.eql(mapping.switch_type, switch_type) and std.meta.eql(mapping.switch_state, switch_state)) {
```

Comment:
I don't think you need `std.meta.eql()` for `switch_type`, it's an enum so `==` should work fine.